src/repository.cc

```diff
@@ -326,6 +328,47 @@ NAN_METHOD(Repository::GetStatus) {
   }
 }
 
+NAN_METHOD(Repository::GetStatusForPaths) {
+  Nan::HandleScope scope;
+  Local<Object> result = Nan::New<Object>();
+  std::map<std::string, unsigned int> statuses;
+  git_status_options options = GIT_STATUS_OPTIONS_INIT;
+  options.flags = GIT_STATUS_OPT_INCLUDE_UNTRACKED |
+                  GIT_STATUS_OPT_RECURSE_UNTRACKED_DIRS |
+                  GIT_STATUS_OPT_DISABLE_PATHSPEC_MATCH;
+
+  Array *pathsArg = Array::Cast(*info[0]);
+  unsigned int pathsLength = pathsArg->Length();
+  char *path = NULL;
+  char **paths = reinterpret_cast<char **>(malloc(pathsLength * sizeof(path)));
+  for (unsigned int i = 0; i < pathsLength; i++) {
+    String::Utf8Value utf8Path(pathsArg->Get(i));
+    path = strdup(*utf8Path);
```

Comment:
That's a really good question. The Internet indicates that `strdup` is POSIX but not Standard. That said, it builds fine on my Windows VM. If we find this is a problem I'll come back and fix this.